testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1718,6 +1719,26 @@ public void testAddColumn()
         assertFalse(getQueryRunner().tableExists(getSession(), tableName));
     }
 
+    @Test
+    public void testAddColumnWithComment()
+    {
+        if (!hasBehavior(SUPPORTS_ADD_COLUMN)) {
+            assertQueryFails("ALTER TABLE nation ADD COLUMN test_add_column_comment varchar", "This connector does not support adding columns");
+            return;
+        }
+
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE));
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_add_column_comment_", "(a_varchar varchar)")) {
+            String tableName = table.getName();
+
+            assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN b_varchar varchar COMMENT 'test new column comment'");
+            if (hasBehavior(SUPPORTS_ADD_COLUMN_WITH_COMMENT)) {
+                assertThat(getColumnComment(tableName, "b_varchar")).isEqualTo("test new column comment");
+            }
```

Comment:
else (when `! hasBehavior(SUPPORTS_ADD_COLUMN_WITH_COMMENT)`)  the ALTER TABLE should  have failed

Reply:
I don't fully understand what you mean.

If the connector does support adding columns without comment it will simply ignore the comment provided, but the `ADD COLUMN` statement will nevertheless succeed.

Reply:
> If the connector does support adding columns without comment it will simply ignore the comment provided

This is a bug, see https://github.com/trinodb/trino/issues/11348 (talks about table comments, but same story)

Reply:
I have updated the PR to reflect this new aspect.
In case that the connector does not support adding a column with comment, it will fail to execute the `ADD COLUMN` task when the comment is present.